### DISCUSSION TITLE
not syncing all mybooks

### DISCUSSION THREAD

**Discussion by Stefan-59** [hadynz/obsidian-kindle-plugin#194] [Q&A]
Hi there,

I have sent an ebook (a PFD converted with calibre as epub) to my Kindle library and read and annotated that book with the Kindle app on my Android tablet. The annotations are synchronized in the Kindlereader between the tablet my phone.

When I start the Kindle sync from Obsidian, it says:
![image](https://user-images.githubusercontent.com/61938696/188433841-28ffc563-b0a6-4f9a-94ea-b895d5ee5ce8.png)

but when syncing, only 14 titles are synced (the new book is NOT synced):
![image](https://user-images.githubusercontent.com/61938696/188433997-9598d8f8-4767-4728-97e4-2efb9b7bd750.png)

What is wrong?
I already signed OFF and ON again with my Amazon account, uninstalled and reinstalled the plugin, no difference.

Edit:
I'd like to try importing the 'My Clippings.txt' but I cannot find such a file on my Android device.
Where would that be stored? Does this file only exist on a hardware Kindle reader?

**Comment by hadynz** [comment]
The plugin supports two methods of syncing - each with their own constraints:

1. Syncing via Amazon cloud - _constraint_: only works for books purchased via Amazon (aka viewable on https://read.amazon.com/notebook)

2. Syncing via `My Clippings.txt` - _constraint_: this files only exists on a physical Kindle device, and will contain all content regardless if it's been purchased from Amazon or not

So to answer your questions:

> but when syncing, only 14 titles are synced (the new book is NOT synced):

That is expected. You are trying to sync a non-Amazon book from Amazon cloud. It won't exist.

> I'd like to try importing the 'My Clippings.txt' but I cannot find such a file on my Android device.

That's correct. This file won't exist on an Android or an iOS device running the Kindle app. It requires a physical Kindle reader

Hope this explanation clarifies the issue at hand.

**Comment by Stefan-59** [comment]
Thank you for your immediate answer!

If I get it right, there's NO way in syncing my highlights into my Obisidan vault with this plugin. That's too sad. :-(

On the other hand, I can export the highlights manually from within Kindle in different formats ('APA', 'Chicago style', 'MLA', 'none').

Would you have an advice, which one to use to at least copy / paste the highlights in an as similar as possible format into my Kindle highlights folder manually?

**Comment by Stefan-59** [comment]
Thank you for your immediate answer!

If I get it right, there's NO way in syncing my highlights into my Obisidan vault with this plugin?

If so, I can export the highlights manually from within Kindle in different formats ('APA', 'Chicago style', 'MLA', 'none').

Would you have any advice, which one to use to at least copy / paste the highlights in an as similar as possible format into my Kindle highlights folder manually?
Or ideally how I would convert the downloaded (html) document into your Kindle highlight-format?

**Comment by campsean** [comment]
The same thing is happening to me, but I did purchase all my books through Amazon.  Right now, the add-in is pulling about 300+ books, but I've got closer to 1,000.

Any ideas as to what's up?

**Comment by hadynz** [comment]
> If I get it right, there's NO way in syncing my highlights into my Obisidan vault with this plugin?
> If so, I can export the highlights manually from within Kindle in different formats ('APA', 'Chicago style', 'MLA', 'none').

Now that you mention it. There is possibly a way; albeit possibly inconvenient for now...

All the Kindle apps (iOS and Android) allow you to export the highlights of an individual book (I don't believe there is a bulk export option yet in them). So one possibly idea is that you export the highlights from your Android Kindle app to your email (`none` format) and then you manually import it into Obsidian via the plugin (I will need to implement this functionality).

There is definitely at least one request (https://github.com/hadynz/obsidian-kindle-plugin/issues/146) that I've heard for doing something for Android.

What's your thoughts on the practicality of this option given the number of books you have?

**Reply by campsean** [reply]
For me at least that would be pretty cumbersome, since that would mean I would have to perform that manual action 700+ times in the iOS app, then email the highlights, then copy and paste them into Obsidian, etc.

Mostly I'm just curious as to why it stops before going through all the books, as it seems somewhat random to stop at around 300+ and not go farther.

**Reply by hadynz** [reply]
We might be talking about slightly different things.

>  then copy and paste them into Obsidian, etc

The data exported from the Kindle app might not be in the exact format that you want in your vault. Just like the Kindle Highlights plugin imports `My Clippings.txt` and transforms it, I can add a feature that would import your exported file and transform it in a consistent manner.

It will continue to be cumbersome until Amazon comes up with a way to bulk export.

> Mostly I'm just curious as to why it stops before going through all the books, as it seems somewhat random to stop at around 300+ and not go farther.

I believe you are referring to a different problem. Are you suggesting that you have purchased over a 1000 books and when you import through the Amazon cloud, the plugin will only import 300? Can you login to https://read.amazon.com/notebook and confirm that you can see all 1,000 of your books and not just 300?
